controller/main.go

```diff
@@ -98,34 +99,28 @@ func (c *controller) deleteBalancer(l log.Logger, name string) {
 	}
 }
 
-func (c *controller) SetConfig(l log.Logger, cfg *config.Config) k8s.SyncState {
+func (c *controller) SetConfig(l log.Logger, cfg *config.Config) controllers.SyncState {
 	level.Debug(l).Log("event", "startUpdate", "msg", "start of config update")
 	defer level.Debug(l).Log("event", "endUpdate", "msg", "end of config update")
 
 	if cfg == nil {
 		level.Error(l).Log("op", "setConfig", "error", "no MetalLB configuration in cluster", "msg", "configuration is missing, MetalLB will not function")
-		return k8s.SyncStateErrorNoRetry
+		return controllers.SyncStateErrorNoRetry
 	}
 
 	if err := c.ips.SetPools(cfg.Pools); err != nil {
 		level.Error(l).Log("op", "setConfig", "error", err, "msg", "applying new configuration failed")
-		return k8s.SyncStateError
+		return controllers.SyncStateError
 	}
 	c.config = cfg
-	return k8s.SyncStateReprocessAll
-}
-
-func (c *controller) MarkSynced(l log.Logger) {
-	c.synced = true
-	level.Info(l).Log("event", "stateSynced", "msg", "controller synced, can allocate IPs now")
+	return controllers.SyncStateReprocessAll
 }
 
 func main() {
 	var (
-		port            = flag.Int("port", 7472, "HTTP listening port for Prometheus metrics")
-		config          = flag.String("config", "config", "Kubernetes ConfigMap containing MetalLB's configuration")
-		namespace       = flag.String("namespace", os.Getenv("METALLB_NAMESPACE"), "config / memberlist secret namespace")
-		kubeconfig      = flag.String("kubeconfig", "", "absolute path to the kubeconfig file (only needed when running outside of k8s)")
+		port      = flag.Int("port", 7472, "HTTP listening port for Prometheus metrics")
+		namespace = flag.String("namespace", os.Getenv("METALLB_NAMESPACE"), "config / memberlist secret namespace")
+		// kubeconfig      = flag.String("kubeconfig", "", "absolute path to the kubeconfig file (only needed when running outside of k8s)")
```

Comment:
Why it's commented?

Reply:
it's a result of my ramblings with controller-runtime, turns out the runtime looks for kubeconfig by itself and the parameter is no needed. I'll remove it.